tailnet/test/integration/network.go

```diff
@@ -586,24 +582,37 @@ func createVethPair(parentVethName, peerVethName string) error {
 }
 
 // setVethNetNS moves the veth interface to the specified network namespace.
-func setVethNetNS(vethName string, netNSFd int) error {
+func setVethNetNS(vethName string, ns NetNamespace) error {
 	veth, err := netlink.LinkByName(vethName)
 	if err != nil {
 		return xerrors.Errorf("LinkByName(%q): %w", vethName, err)
 	}
 
-	err = netlink.LinkSetNsFd(veth, netNSFd)
+	raw, err := ns.File.SyscallConn()
```

Comment:
Was this causing some issue? I don't understand the difference to the existing code

Reply:
I didn't see any issues, but doing it the original way is racy because Go doesn't guarantee that `File.Fd()` returns a file descriptor that is stable.  Go reuses file descriptors. It's generally safer to do syscalls on files via https://pkg.go.dev/syscall#RawConn